File: backend/internal/model/news.go
package model

import "time"

// News represents a steel industry news article.
type News struct {
	ID          uint      `gorm:"column:id;primaryKey" json:"id"`
	Title       string    `gorm:"column:title" json:"title"`
	Summary     string    `gorm:"column:summary" json:"summary"`
	Content     string    `gorm:"column:content" json:"content"`
	Source      string    `gorm:"column:source" json:"source"`
	SourceURL   string    `gorm:"column:source_url" json:"source_url"`
	Category    string    `gorm:"column:category" json:"category"`
	PublishedAt time.Time `gorm:"column:published_at" json:"published_at"`
	CreatedAt   time.Time `gorm:"column:created_at" json:"created_at"`
}

// TableName returns the database table name for News.
func (News) TableName() string {
	return "news"
}

File: steel-agent-web/src/app/types/news.ts
// ============================================================
// 资讯新闻相关 TypeScript 类型定义
// 对应后端 /api/v1/news/* 接口
// ============================================================

/** 资讯详情 */
export interface NewsDetail {
  id: string;
  title: string;
  content: string;
  source: string;
  source_url: string;
  category: string;
  published_at: string;
}
